espnet/nets/chainer_backend/decoders.py

```diff
@@ -43,6 +45,17 @@ def __init__(self, eprojs, odim, dlayers, dunits, sos, eos, att, verbose=0,
         self.lsm_weight = lsm_weight
         self.sampling_probability = sampling_probability
 
+    def rnn_forward(self, ey, z_list, c_list, z_prev, c_prev):
+        if self.dtype == "lstm":
+            c_list[0], z_list[0] = self.rnn0(c_prev[0], z_prev[0], ey)
+            for l in six.moves.range(1, self.dlayers):
+                c_list[l], z_list[l] = self['rnn%d' % l](c_prev[l], z_prev[l], z_list[l - 1])
+        else:
+            z_list[0] = self.rnn0(z_prev[0], ey)
```

Comment:
it will be better if you add something like this:
```
if z_prev[0] is None:
                xp = self.xp
                with cuda.get_device_from_id(self._device_id):
                    z_prev[0] = variable.Variable(
                        xp.zeros((ey.shape[0], self.dunits), dtype=ey.dtype))
```
because GRU does not prepare a zeros array as Stateless LSTM does.

Reply:
Thanks for the fix ! That's a strange design choice by chainer to have different behaviors for StatelessLSTM and StatelessGRU though.